### ISSUE TITLE
Debugging test with Node.js

### ISSUE THREAD

**Issue by Jarlakxen** [scala-js/scala-js#3126]
I'm porting a Scala project to Scala.js and some of the test are failing with Scala.js and I need to trace what's going on. This application needs to run as a Node.js app, so I need to run and debug the tests in a Node.js enviroment.
I tried by adding `jsEnv in Test := new org.scalajs.jsenv.nodejs.NodeJSEnv(org.scalajs.jsenv.nodejs.NodeJSEnv.Config().withArgs(List("--inspect-brk")))` in the build.sbt but when I run test I get this in the chrome inspector:
```
@scalajs-test-framework-detector:org.scalacheck.ScalaCheckFramework    VM84 [stdin]:28
@scalajs-test-framework-detector:                                                                                    VM84 [stdin]:28
@scalajs-test-framework-detector:org.scalatest.tools.Framework                     VM84 [stdin]:28 
@scalajs-test-framework-detector:
```
The execution is not stopping and I can't see the source code in the chrome inspector. I made a small project try debugging a simple test: https://github.com/Jarlakxen/scalajs-node-debugging
If I run `sbt test` I'm expecting to see the chrome inspector show up stopped in the first js line.

**Comment by sjrd** [comment]
When I try running `test` inside the project you provide, I just get a
```
[debug] Starting process: node --inspect-brk
java.io.IOException: Stream closed
        at java.lang.ProcessBuilder$NullOutputStream.write(ProcessBuilder.java:433)
        at java.io.OutputStream.write(OutputStream.java:116)
        at java.io.BufferedOutputStream.flushBuffer(BufferedOutputStream.java:82)
        at java.io.BufferedOutputStream.flush(BufferedOutputStream.java:140)
        at java.io.FilterOutputStream.close(FilterOutputStream.java:158)
        at org.scalajs.jsenv.ExternalJSEnv$AbstractExtRunner.pipeVMData(ExternalJSEnv.scala:97)
        ...
```
Is there something I need to install/launch separately?

**Comment by Jarlakxen** [comment]
Mmh... that's weird. Which version of Node.js do you have?
I'm using 8, but with 7 should be the same.

**Comment by sjrd** [comment]
Ah, I have 6.9.1. I'll retry later with a more recent version.

**Comment by Jarlakxen** [comment]
Oh! Sorry I forgot to mention the node version. Between 6 and 7 they changed `--inspect --debug-brk` to `--inspect-brk`

**Comment by sjrd** [comment]
This seems to work very well with Scala.js 0.6.23 and Node.js v10.0.0. I suspect it was fixed in 0.6.22 thanks to #3122.

Use
```
parallelExecution in Test := false
```
in sbt settings to prevent Scala.js from using multiple Node.js processes (before 0.6.22 that was not good enough because Scala.js would still use 3 processes with that config; not anymore).

You can then debug with the Chrome DevTools, although *at first* you won't see the sources, because they are dynamically loaded with a `require`. I recommend calling
```scala
scala.scalajs.js.special.debugger()
```
at the beginning of the test you want to debug. That works well.
